js/index.js

```diff
@@ -43,27 +43,82 @@ function renderPepperoni() {
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(shroom => {
+    if (state.mushrooms) {
+      shroom.style.visibility = 'visible';
+    } else {
+      shroom.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(pepper => {
+    if (state.greenPeppers) {
+      pepper.style.visibility = 'visible';
+    } else {
+      pepper.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce-white').forEach(x => {
+    if (state.whiteSauce) {
+      x.style.visibility = 'visible';
```

Comment:
in this case the visibility option is not working, because the moment you take it out, it disappears the whole element. And what you actually want to do is to add and remove the class white-sauce :)
so in the if statememnts something like this.

```
if(state.whiteSauce){
x.classList.add("Sauce-white")// at the end you put the name of the class inside the add function

}
else{
x.classList.remove("Sauce")
}

```
ALSO! take in account you are selecting whatever element that has .sauce-white so if you remove it then you wont be able to select it again.
So, select only ".sauce" instead
example:
```
document.querySelectorAll(".sauce").forEach((x)=>{whatever})
```